### PR TITLE
Refactor scaling

### PR DESCRIPTION
**Pull request by toaster**
The previous implementation set the scaling to 1.0 at a resolution of 144dpi. That's quite small and unreadable on displays with (much) less dpi like my old 24" screen with 94dpi.

This baseline was changed to 96dpi.

On the mentioned display the resulting scale of 0.98… made the text very blurry. That's why the scaling precision was reduced to 1 so that the effective scale has values like 0.8, 0.9, 1.0, 1.1, 1.2 etc.

The math methods used for the scale computations mostly require `float64` and because there was no requirement to use `float32` anywhere where scale was used, I moved scale, and some other measures that are involved in calculations with scale, to `float64`.

### CODE REVIEW

File: canvas.go [hunk 1]
```diff
@@ -13,8 +13,8 @@ type Canvas interface {
 	Focused() Focusable
 
 	Size() Size
-	Scale() float32
-	SetScale(float32)
+	Scale() float64
```
Comment (andydotxyz) [inline canvas.go]: This is a breaking api change - anyone accessing this function and using the value could get broken code as a result

File: canvas/circle.go [hunk 2]
```diff
@@ -17,7 +17,7 @@ type Circle struct {
 
 	FillColor   color.Color // The circle fill colour
 	StrokeColor color.Color // The circle stroke colour
-	StrokeWidth float32     // The stroke width of the circle
```
Comment (andydotxyz) [inline canvas/circle.go]: It looks like the stroke changes are not part of this change?

**Comment by andydotxyz** [comment]
Not sure the new values work - this is my laptop on a 3200x1800 screen and the content is huge...
Screenshot from 2019-09-07 22-18-31

**Comment by andydotxyz** [comment]
Before the path the scale was 1.95 and the size seemed about right

**Comment by andydotxyz** [comment]
Playing around with this more I wonder if actually what we should do is have platform specific "auto" and `if runtime.GOOS == "darwin" { return 1.0 }`...

**Comment by andydotxyz** [comment]
I propose #454 instead. It uses the same clever rounding to 1 decimal place to avoid fuzzing but it defaults to 1.0 as it seems most systems prefer this.

On linux you set FYNE_SCALE="auto" and on mobile we don't use this code anyway!

**Comment by andydotxyz** [comment]
should this be closed now?

**Comment by toaster** [comment]
yes…
